lib/usbd/usbd_ep0.c

```diff
@@ -340,13 +340,13 @@ standard_get_descriptor_string(usbd_device *dev,
 
 	if (!index) {
 		/* Build the list of available langauge strings */
-		used = build_available_lang(dev, sd->wData, max_count);
+		used = build_available_lang(dev, (uint16_t*)sd->wData, max_count);
 	} else {
 		/* Search for the UTF-8 string and convert it to UTF-16 */
 		const uint8_t *utf8_str;
 		utf8_str = search_utf8_string(dev, arg->setup->wIndex, index - 1);
 		used = (utf8_str == NULL) ? -1 :
-			utf8_to_utf16(utf8_str, sd->wData, max_count);
+			utf8_to_utf16(utf8_str, (uint16_t*)sd->wData, max_count);
```

Comment:
No, that is not:  sd->wData is of type "packed" uint16_t*, since it got from packed struct. 
maybe there should be some comments. But , i guess, the temporary buffer preserve.buf - always aligned, so better to dropoff "packed" attribute, instead rewrite utf8_to_utf16 and search_utf8_string capable work with unaligned utf16 strings.